docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,160 @@
+---
+title: Persisted Queries
+---
+
+Apollo supports two separate but related features called **persisted queries** and **automated persisted queries** (APQs).
+With both features, clients can execute a GraphQL operation by sending an operation's ID instead of the entire string. An operation's ID is the SHA256 hash of the full operation string. Querying by ID can significantly reduce latency and bandwidth usage for very large operation strings.
+
+## Differences between persisted queries and APQ
+
+The persisted queries feature requires operations to be preregistered in a **persisted query list** (**PQL**). 
+This allows the PQL to act as a safelist of trusted operations made by your first-party apps. As such, persisted queries is a security feature as much as a performance one.
+
+With APQs, if the server can't find the operation ID the client provides, the server returns an error indicating that it needs the full operation string. If the Apollo iOS client receives this error, it automatically retries the operation with the full operation string.
+
+If you _only_ want to improve request latency and bandwidth usage, APQ addresses your use case. If you _also_ want to secure your supergraph with operation safelisting, you should preregister trusted operations in a PQL.
+
+For more details on differences between persisted queries and APQ, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries#differences-from-automatic-persisted).
+
+### Requirements
+
+Persisted queries is currently in [preview](/resources/product-launch-stages#preview) and has the following requirements:
+- Apollo iOS (v1.3.1+)
+- [Apollo Router](/router) (v1.25.0+)
+- [GraphOS Enterprise plan](/graphos/enterprise/)
+
+You can use APQ with the following versions of Apollo iOS, Apollo Router, and Apollo Server:
+- Apollo iOS (v1.0.0+)
+- [Apollo Router](/router) (v0.1.0+)
+- [Apollo Server](/apollo-server/) (v1.0.0+)
+
+## Implementation steps
+
+Both persisted queries and APQ require you to configure code generation and how your client makes requests. The rest of this article details these configurations.
+
+Persisted queries also requires you to create and link a PQL and configure your router. For more information on the other configuration aspects of persisted queries, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries).
+
+### 1. Configure code generation
+
+The code generation requirements for persisted queries and APQ are the same: your operations need a SHA256 hash identifier included in the generated operation objects. To configure code generation accordingly, make the following changes in your `ApolloCodegenConfiguration` JSON file:
+
+1. In the `output` section of your configuration, set the [`operationManifest`](./../code-generation/codegen-configuration#operation-manifest) with:
```

Comment:
If you want to generate operation manifest every time you do codegen, this is how you do that, but actually, we recommend that you only generated the manifest as needed using the CLI command as noted below on line 67 of this document.

It looks like we actually need to make an addition to the `generate-operation-manifest` command to make this possible though. So this documentation is accurate to the current state of things. Speaking with @BobaFetters about making the addition so we can update these docs to reflect what we want users to be able to do.

---

You really only need the `operationManifest` for `APQ` if you are also using the safe-listing capabilities or choosing to pre-register your operations with APQs. 

The messaging up above explaining the difference between `APQ` and `persistedQueries` doesn't mention anything about pre-registering documents when using `APQs. It seems to indicate that pre-registering queries is really only a feature of `persistedQueries`.

@glasser Do we still want to even be mentioning the ability to pre-register queries in the APQ world? Or are we trying to remove references to that feature and discourage its use since it was buggy and `persistedQueries` is replacing it?

---

Once we've made the change, I'm wondering if we could make this more clear by organizing this as follows:

It could be 

```markdown
### 1. Generate an operation manifest
{Explain what the operation manifest is for.}
{Explain that the manifest is needed for persisted queries and optional for APQs if you want to pre-register them?}

{Explain the primary way of generating operation manifest, via the CLI command with the `--path` and `--version` arguments}

{Explain the alternative way of adding the `output.operationManifest` to codegen config. Which will re-generate the manifest everytime you run the `generate` command.}

### 2. Configure generated operation models
{Insert the section on `options.operationDocumentFormat` from lines 54-63.}

### 3. Enable persisted queries on `ApolloClient`
{Insert Existing Section}
```

Reply:
PR #3172 is up which adds the desired functionality. We should now change these docs as indicated above.

Reply:
> Or are we trying to remove references to that feature and discourage its use since it was buggy and persistedQueries is replacing it?

I'd imagine we'd prefer folks to use `persistedQueries` over APQs, so will keep the intro as it is. (I.e. not mention you _can_ preregister queries with APQ.)

Re: removing all mentions of `apq`... not sure. Will bring up with product / PMM.

Reply:
> Do we still want to even be mentioning the ability to pre-register queries in the APQ world?

I would vote against mentioning that. I actually don't think you need to pre-register anything with APQ. APQ are completely standalone and don't need pre-registering. This is the beauty of it. On Kotlin, we have an option to generate a manifest in the "old apollo-cli format" (we call it operationOutput) but to my knowledge this isn't used at all by APQ. This was for users who implemented their own persisted queries manually.